src/metabase/models/card.clj

```diff
@@ -521,6 +524,10 @@
       pre-update
       populate-query-fields
       maybe-populate-initially-published-at
+      ;; TODO: this should go in after-update once camsaul/toucan2#129 is fixed
+      ;; It's at the end for now so that all the before-update validations have a chance to run
+      ;; TODO the Second: No reason this couldn't be async, especially once it's in the after-update
```

Comment:
I don't think it should be async, though. Updating should be atomic; that's why before-update -> update -> after-update is executed in a transaction. Moving this out of the transaction will break that promise.